LeaderboardIndexTest::parallel sometimes fails with an uncaught exception

A handful of PRBs have failed with an error along the lines of:

```
15:40:16 com.apple.foundationdb.record.provider.foundationdb.leaderboard.LeaderboardIndexTest > parallel() FAILED
15:40:16     java.lang.AssertionError: 
15:40:16     Expected: an empty collection
15:40:16          but: <[com.apple.foundationdb.record.provider.foundationdb.FDBExceptions$FDBStoreTransactionConflictException: Transaction not committed due to conflict with another transaction]>
15:40:16         at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
15:40:16         at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:8)
15:40:16         at com.apple.foundationdb.record.provider.foundationdb.leaderboard.LeaderboardIndexTest.parallel(LeaderboardIndexTest.java:1165
```

It looks like (from running it) that we expect to get some conflicts, so this error is presumably from hitting the maximum retry limit (sometimes).

--- a/fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/leaderboard/LeaderboardIndexTest.java
+++ b/fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/leaderboard/LeaderboardIndexTest.java
@@ -61,6 +61,8 @@ import com.apple.test.Tags;
 import com.google.common.primitives.Longs;
 import com.google.protobuf.Message;
 import org.hamcrest.Matchers;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Tag;
 import org.junit.jupiter.api.Test;
@@ -89,8 +91,26 @@ import static org.junit.jupiter.api.Assertions.assertTrue;
 public class LeaderboardIndexTest {
     FDBDatabase fdb;
     FDBStoreTimer metrics;
+    private static int oldMaxAttempts;
+    private static final int TEST_MAX_ATTEMPTS = 100;
+    private static long oldMaxDelay;
+    private static final long TEST_MAX_DELAY = 10L;
 
     private static final boolean TRACE = false;
+
+    @BeforeAll
+    public static void initializeRetryPolicy() {
+        oldMaxAttempts = FDBDatabaseFactory.instance().getMaxAttempts();
+        FDBDatabaseFactory.instance().setMaxAttempts(TEST_MAX_ATTEMPTS);
+        oldMaxDelay = FDBDatabaseFactory.instance().getMaxDelayMillis();
+        FDBDatabaseFactory.instance().setMaxDelayMillis(TEST_MAX_DELAY);
+    }
+
+    @AfterAll
+    public static void resetRetryPolicy() {
+        FDBDatabaseFactory.instance().setMaxAttempts(oldMaxAttempts);
+        FDBDatabaseFactory.instance().setMaxDelayMillis(oldMaxDelay);
+    }
 
     @BeforeEach
     public void getFDB() {

